lua/telescope/utils.lua

```diff
@@ -147,6 +147,37 @@ utils.path_tail = (function()
   end
 end)()
 
+utils.is_path_hidden = function(opts)
+  local path_display = utils.get_default(opts.path_display, require('telescope.config').values.path_display)
+
+  return path_display == nil or path_display == "hidden" or
+    type(path_display) ~= "table" or vim.tbl_contains(path_display, "hidden")
+end
+
+utils.transform_path = function(opts, path)
+  if utils.is_path_hidden(opts) then
+    return ''
+  end
+
+  local path_display = utils.get_default(opts.path_display, require('telescope.config').values.path_display)
+  local transformed_path = path
+
+  if vim.tbl_contains(path_display, "tail") then
+    transformed_path = utils.path_tail(transformed_path)
+  else
+    if not vim.in_fast_event() and not vim.tbl_contains(path_display, "absolute") then
+      local cwd = vim.fn.expand(opts.cwd or vim.fn.getcwd())
+      transformed_path = pathlib.make_relative(transformed_path, cwd)
+    end
```

Comment:
I am pretty sure this is not optimal. First of all `vim.loop.cwd()` is always better than `vim.fn.getcwd()` because it can run in an fast event. Second you don't need to expand `cwd` so you could do this:
- `if not opts.cwd` then we can always do `local cwd = vim.loop.cwd()`
- `if opts.cwd` then we need to make sure that we are not `in_fast_event`. if not then `vim.fn.expand`

Reply:
* +1 on `vim.loop.cwd()`
* I think I get what you mean for the other points. Pushing the change